From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <76615486+NONPLAYT@users.noreply.github.com>
Date: Sun, 16 Jul 2023 13:51:26 +0300
Subject: [PATCH] Divine: Fix MC-93018

You can find the original code on https://github.com/DivineMC/DivineMC

Original post on Mojira: https://bugs.mojang.com/browse/MC-93018

diff --git a/src/main/java/net/minecraft/world/entity/animal/Wolf.java b/src/main/java/net/minecraft/world/entity/animal/Wolf.java
index a8193ef23763a11016b9ac8c7dd55b9e240d6039..6ada2a1bc497588b48022c60934bb363d762db70 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Wolf.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Wolf.java
@@ -768,17 +768,15 @@ public class Wolf extends TamableAnimal implements NeutralMob, VariantHolder<Hol
 
     @Override
     public boolean canMate(Animal other) {
+        // DivineMC start - Fix MC-93018
         if (other == this) {
             return false;
-        } else if (!this.isTame()) {
-            return false;
-        } else if (!(other instanceof Wolf)) {
+        } else if (!(other instanceof Wolf wolf)) {
             return false;
         } else {
-            Wolf entitywolf = (Wolf) other;
-
-            return !entitywolf.isTame() ? false : (entitywolf.isInSittingPose() ? false : this.isInLove() && entitywolf.isInLove());
+            return !wolf.isInSittingPose() && (this.isInLove() && wolf.isInLove());
         }
+        // DivineMC end
     }
 
     public boolean isInterested() {
